### PR TITLE
Support for software/virtual sensors

### PR DESCRIPTION
**Pull request by aleksamagicka**
(Initial) Support for virtual software sensors of the Octo, Quadro, Farbwerk 360 and D5 Next.

TODO before merging:

- [x] Read and expose virtual sensors from the sensors report
- [ ] Implement writing to sysfs entries and saving them to the device (need to RE offsets)
- [ ] Explain this in kernel and RE docs (those need to be redone)

### CODE REVIEW

File: aquacomputer_d5next.c [hunk 1]
```diff
@@ -371,6 +401,8 @@ static umode_t aqc_is_visible(const void *data, enum hwmon_sensor_types type, u3
 	case hwmon_temp:
 		if (channel < priv->num_temp_sensors)
 			return 0444;
+		if (channel < priv->num_virtual_temp_sensors)
```
Comment (leoratte) [inline aquacomputer_d5next.c:404]: should be `if (channel < priv->num_temp_sensors + priv->num_virtual_temp_sensors)` to accommodate for all virtual sensors

File: aquacomputer_d5next.c [hunk 2]
```diff
@@ -498,7 +530,10 @@ static int aqc_read_string(struct device *dev, enum hwmon_sensor_types type, u32
 
 	switch (type) {
 	case hwmon_temp:
-		*str = priv->temp_label[channel];
+		if (channel < priv->num_temp_sensors)
+			*str = priv->temp_label[channel];
+		else
+			*str = priv->virtual_temp_label[channel - priv->num_temp_sensors - 1];
```
Comment (leoratte) [inline aquacomputer_d5next.c:536]: remove  `-1` as channel starts at 0 and so the first virtual sensor would get virtual_temp_label[-1]

**Review by leoratte — CHANGES_REQUESTED** [review]

**Comment by aleksamagicka** [comment]
Thanks, fixed these oversights. I distinctly remember writing that and feeling something was off :sweat_smile:

**Comment by aleksamagicka** [comment]
Note: it looks like virtual temp sensors automatically expire on the device after 10-15 seconds of no further updates.

**Pull request merged**